xtask/src/main.rs.liquid

```diff
@@ -83,21 +75,13 @@ fn check_usage() {
 //////////////////////////////////// Checks ////////////////////////////////////
 
 fn check_commits(ctx: &mut Context) {
-    let is_pull_request =
-        env::var_os("IS_PULL_REQUEST").is_some_and(|val| val == "true");
-
-    let commits = if is_pull_request {
-        Some(String::from("HEAD~..HEAD^2"))
-    } else {
-        let branch = get_current_branch();
-
-        if ["main", "develop"].contains(&branch.as_str()) {
-            None
-        } else {
-            let merge_base = get_merge_base("origin/develop");
-            Some(format!("{merge_base}..HEAD"))
-        }
-    };
+    let commits = env::var_os("IS_PULL_REQUEST")
+        .is_some_and(|val| val == "true")
+        .then_some(String::from("HEAD~..HEAD^2"))
+        .or(
+            (!["main", "develop"].contains(&get_current_branch().as_str()))
+                .then(|| format!("{}..HEAD", get_merge_base("origin/develop"))),
+        );
```

Comment:
**issue:** I find this less readable. It’s fancier and definitely *looks* more elegant, however it is a bit more difficult to get a mental model of the different cases from it.

In the original code, the three cases are quite visible:

1. GitHub PR => `HEAD~..HEAD^2`
2. `main` or `develop` => Do not check
3. Other branches => `<merge_base>..HEAD`

They are not in the new code.

I’d suggest to stick to the current one, except if you find an elegant way of showing those three cases.